### ISSUE TITLE
Add how to publish projects on the catalog

### ISSUE THREAD

**Issue by SergeStinckwich** [pharo-open-documentation/pharo-wiki#282]
Use the following conversation on Discord:

Stepharo: So how do we publish projects on the catalog?
[9:30 PM]  astares: @Stepharo 

It works the same as it worked over the last years and Pharo versions by providing a config on SmalltalkHub. But now if you have a GitHub project you just point to BaselineOf... at GitHub and load that from the config. 

So basically you write a ConfigurationOfMyKillerApp class and publish it as an MCZ to a MetaRepoPharoXX on SmalltalkHub. Every 24hour there is a job that is grabbing the configs there and creating the JSON file http://catalog.pharo.org/catalog/json  which is then retrieved by the catalog client UI in the image.

There are meta-repos for Pharo 3.0 up to 9.0
    http://www.smalltalkhub.com/#!/~Pharo/MetaRepoForPharo30
    http://www.smalltalkhub.com/#!/~Pharo/MetaRepoForPharo40
    http://www.smalltalkhub.com/#!/~Pharo/MetaRepoForPharo50
    http://www.smalltalkhub.com/#!/~Pharo/MetaRepoForPharo60
    http://www.smalltalkhub.com/#!/~Pharo/MetaRepoForPharo70
    http://www.smalltalkhub.com/#!/~Pharo/MetaRepoForPharo80
   http://www.smalltalkhub.com/#!/~Pharo/MetaRepoForPharo90

(yes I know - SmalltalkHub is not fashionable anymore - but it worked and served well so far nontheless and it will help us until we have full replacement)
[9:31 PM]  astares: Here is an example:
[9:34 PM]  astares:
CatalogProject createRepositoryURLtoNameMappings

The catalog URL is configurable in the settings - so you can modify it at startup if you like to even point to a different JSON that is used
CatalogSettings catalogSourceURL
[9:35 PM]  astares: As of today it points to <http://catalog.pharo.org/catalog/json >
[9:37 PM]  astares: _
[9:45 PM]  astares: Let me explain you with an example: 
 - As you know I migrated the XML support to GitHub
 - for instance there is this GitHub project https://github.com/pharo-contributions/XML-XMLWriter including a baseline and it is loadable via script 

So if I want to load latest I need to load
Metacello new
    baseline: 'XMLWriter';
    repository: 'github://pharo-contributions/XML-XMLWriter/src';
    load.

which is equal to
Metacello new
    baseline: 'XMLWriter';
    repository: 'github://pharo-contributions/XML-XMLWriter:master/src';
    load.

 - but additional I tagged in the master branch a version defining a release version "2.9.0" in the master branch (like in Iceberg and any other GitHub hosted released project)

Metacello new
    baseline: 'XMLWriter';
    repository: 'github://pharo-contributions/XML-XMLWriter:2.9.0/src';
    load.

 
Now in my ConfigurationOfXMLWriter class I want to define a (Metacello) version pointing to this specific GitHub tagged version. So I write:

version290: spec 
    <version: '2.9.0'>

    spec for: #'common' do: [ 
        spec
            baseline: 'XMLWriter' with: [ 
                spec 
                    className: 'BaselineOfXMLWriter';
                    repository: 'github://pharo-contributions/XML-XMLWriter:2.9.0/src' ]]

and for Monticello define a 

stable: spec
    <symbolicVersion: #'stable'>
    
    spec for: #'common' version: '2.9.0'

as we did in the old days and then I upload the config into the MetaRepo of the supported Pharo version
[9:52 PM]  astares: This is only one model you can manage the differences.

Another one is to do it like I did in ConfigurationOfDesktopManager where I defined a stable (Monticello) release depending on Pharo version:

stable: spec
    <symbolicVersion: #'stable'>

    spec for: #'pharo5.x' version: '0.2.0'.
    spec for: #'pharo6.x' version: '6.0'.
    spec for: #'pharo7.x' version: '7.0'.
        ...

and pointing to a branch responsible for the specific Pharo version
pharo7: spec
    <version: '7.0'>

    spec for: #'common' do: [ 
        spec
            baseline: 'DesktopManager' with: [ 
                spec 
                    className: 'BaselineOfDesktopManager';
                    repository: 'github://astares/Pharo-DesktopManager:pharo7/src' ]]

By having a "pharo7" and "pharo8" branch I can use Git to manage differences if I like or need.
[9:55 PM]  astares: So basically we do not have to really get rid of Catalog within the image - we have to possibly get rid of SmalltalkHub and the additional Configs.
Basically we only need a way to build the JSON for a specific Pharo version - we can host it on GitHub.

**Comment by jecisc** [comment]
I have the impression that the catalog is not part of Pharo anymore.
I’ll close it for now
